### ISSUE TITLE
pywinauto actionlogger formatting

### ISSUE THREAD

**Issue by meshuggahtas** [pywinauto/pywinauto#538]
I would like to have more control over pywinauto actionlogger but can't seem to get it correctly.

1. actionlogger.enable() - actionlogger prints to sys.stderr (for example I have red colored INFO message):

`2018-08-09 11:45:09,808 INFO: Started notepad.exe application.`

2. actionlogger.ActionLogger.logger.propagate = True - okay, this way I can fit in my original logger to format it in the same way, however now I get double messages.

3. Using a custom class to redirect sys.stderr using https://stackoverflow.com/questions/19425736/how-to-redirect-stdout-and-stderr-to-logger-in-python/31688396 - okay, but this way my original error messages (anything else than pywinauto) also redirected.

`11:54:01.358INFO            pywinauto         log                 @156 Started notepad.exe application.`

So... how should I enable pywinauto actionlogger (to see the mouse movement log messages) and format it without formatting sys.stderr as a side product and getting duplicates?

**Example code where pywinauto will log to sys.stderr:**

```
import logging

file = logging.FileHandler(filename="logtest.txt", mode="w")

logging.basicConfig(level=logging.DEBUG,
                    format="%(asctime)s.%(msecs)03d%(levelname)-15s %(name)-18.17s%(funcName)-20.19s@%(lineno)+3s %(message)s",
                    datefmt="%H:%M:%S",
                    handlers=[file])

log = logging.getLogger("main")

log.info("info")
log.error("error")

from pywinauto import actionlogger, application
actionlogger.enable()
#actionlogger.ActionLogger.logger.propagate = True
log.debug(actionlogger.ActionLogger.logger.handlers)

application.Application(backend="win32").start("notepad.exe")
```

**Result of actionlogger handlers:** `11:54:01.359DEBUG           main              <module>            @ 22 [<StreamHandler <stderr> (DEBUG)>]`

**Comment by vasily-v-ryabov** [comment]
Well, started looking into this...

Yes, default handler for pywinauto logs is `logging.StreamHandler()` which prints to `sys.stderr` according to Python docs. If it's not what you want, just remove this handler and add FileHandler. This code adds your file handler to pywinauto logger, while `basicConfig` call doesn't add it to "main" logger (incorrect `basicConfig` usage?):
```python
import logging

file_handler = logging.FileHandler(filename="logtest.txt", mode="w")

logging.basicConfig(level=logging.DEBUG,
                    format="%(asctime)s.%(msecs)03d%(levelname)-15s %(name)-18.17s%(funcName)-20.19s@%(lineno)+3s %(message)s",
                    datefmt="%H:%M:%S",
                    handlers=[file_handler])

log = logging.getLogger("main")
log.debug(log.handlers)

log.info("info")
log.error("error")

from pywinauto import actionlogger, application
actionlogger.enable()
actionlogger.ActionLogger.logger.propagate = True
stream_handler = actionlogger.ActionLogger.logger.handlers[0]
actionlogger.ActionLogger.logger.removeHandler(stream_handler)
actionlogger.ActionLogger.logger.addHandler(file_handler)
log.debug(actionlogger.ActionLogger.logger.handlers)

application.Application(backend="win32").start("notepad.exe")
```

Currently I don't see anything to do inside pywinauto. Requalifying as question.

**Comment by meshuggahtas** [comment]
It's okay to be a question as I'm still puzzled because of lack of proper knowledge on how to do this:
_(parenthesis: my code and your code)_
- Handler for the program (file / file_handler) which is formatted
- Handler for pywinauto (N/A / stream_handler) which is not formatted
- Getting all from pywinauto logger to my formatted logger

I want my formats (date, time, level, funcname) to be in front of stderrs as well.

My notes so far:
- Maybe the problem is related to this from https://docs.python.org/3.7/howto/logging.html "The default value of raiseExceptions is True. This is because during development, you typically want to be notified of any exceptions that occur. It’s advised that you set raiseExceptions to False for production usage. "
- The pywinauto docs have very small information, only found this page relevant https://pywinauto.readthedocs.io/en/latest/code/pywinauto.actionlogger.html?highlight=logger
- I've tried to use the debugger to found a logger which was _StandardLogger for me (because I imported simply "logger" instead of "actionlogger") but it seems I need actionlogger.

**Comment by vasily-v-ryabov** [comment]
`actionlogger.ActionLogger.logger` is a static member of class `ActionLogger` (therefore it's kind of global variable). Agree that the docs are very minimal. I think we need to improve it.

You also can see how new formatter can be added to this logger: [actionlogger.py, Line 112](https://github.com/pywinauto/pywinauto/blob/master/pywinauto/actionlogger.py#L112).

**Comment by meshuggahtas** [comment]
Ah, rereading your earlier https://github.com/pywinauto/pywinauto/issues/538#issuecomment-453757123 helped me!
```
actionlogger.enable()
actionlogger.ActionLogger.logger.propagate = True
actionlogger.ActionLogger.logger.removeHandler(actionlogger.ActionLogger.logger.handlers[0])
actionlogger.ActionLogger.logger.addHandler(logger.file)
```
Thank you and sorry for wasting your time.

- So I wanted to have a main logger that formats all messages of the program.
- I rerouted sys.stderr as I linked in the OP 3rd point and enabled propagation. 
- I've mistakenly thought that creating the basicConfig in my module will be affect new loggers in the program
- As you mentioned, I just had to remove the handler from ActionLogger and add my own handler.
